### PR TITLE
Add states and reducers

### PR DESCRIPTION
**Pull request by MaximBazarov**

### CODE REVIEW

File: State/DocumentState.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//  DocumentState.swift
+//  ReduxVisualEditor
+//
+//  Created by Maxim Bazarov on 26.01.19.
+//  Copyright © 2019 Unicore. All rights reserved.
+//
+
+import Unicore
+
+struct DocumentState: State {
```
Comment (AlexeyDemedetskiy) [inline State/DocumentState.swift]: upfront?

drop it for now.

File: State/ActionsList.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+//  ActionsList.swift
+//  ReduxVisualEditor
+//
+//  Created by Maxim Bazarov on 26.01.19.
+//  Copyright © 2019 Unicore. All rights reserved.
+//
+
+import Unicore
+
+extension Action {
```
Comment (AlexeyDemedetskiy) [inline State/ActionsList.swift]: It is unicore Action or action list element?

File: State/ActionsLinks.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//
+//  ActionsLinks.swift
+//  ReduxVisualEditor
+//
+//  Created by Maxim Bazarov on 26.01.19.
+//  Copyright © 2019 Unicore. All rights reserved.
+//
+
+import Unicore
```
Comment (AlexeyDemedetskiy) [inline State/ActionsLinks.swift]: Can we not have unicore imports?

File: State/ActionsList.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    typealias Name = String
```
Comment (AlexeyDemedetskiy) [inline State/ActionsList.swift]: Use strong phantom type instead

File: State/DocumentState.swift [hunk 5]
```diff
@@ -0,0 +1,26 @@
+//
+//  DocumentState.swift
+//  ReduxVisualEditor
+//
+//  Created by Maxim Bazarov on 26.01.19.
+//  Copyright © 2019 Unicore. All rights reserved.
+//
```
Comment (AlexeyDemedetskiy) [inline State/DocumentState.swift]: drop comments

File: State/ActionsLinks.swift (hunk 3 above)
Comment (MaximBazarov) [reply State/ActionsLinks.swift]: but how then we're going to mark it as an Action?

File: State/ActionsList.swift (hunk 2 above)
Comment (MaximBazarov) [reply State/ActionsList.swift]: Unicore one

File: State/ActionsLinks.swift [hunk 6]
```diff
@@ -0,0 +1,9 @@
+struct ActionsLinks {
+    let links: [ActionsList.Action: [StatesList.State]]
```
Comment (MaximBazarov) [inline State/ActionsLinks.swift]: Here seems like ActionLinks depends on ActionList?

**Pull request merged**